yarn-project/simulator/src/public/enqueued_call_side_effect_trace.ts

```diff
@@ -346,10 +368,8 @@ export class PublicEnqueuedCallSideEffectTrace implements PublicSideEffectTraceI
     // NOTE: isPending and leafIndex are unused for now but may be used for optimizations or kernel hints later
     this.enforceLimitOnNullifierChecks();
 
-    // TODO(dbanks12): use siloed nullifier instead of scoped once public kernel stops siloing
-    // and once VM public inputs are meant to contain siloed nullifiers.
-    //const siloedNullifier = siloNullifier(contractAddress, nullifier);
-    const readRequest = new ReadRequest(nullifier, this.sideEffectCounter).scope(contractAddress);
+    // we scope with a dummy address because this read request isn't used
+    const readRequest = new ReadRequest(siloedNullifier, this.sideEffectCounter).scope(AztecAddress.zero());
     if (exists) {
```

Comment:
We could probably remove all "read requests" of this sort from the new trace. Would still need to track how many have occurred though to make sure we don't surpass some limit.

Reply:
Could honestly just enforce the limit based on the length of the hints vectors.